### PR TITLE
IF: remove chain_pacemaker and qc_chain related files

### PR DESCRIPTION
**Pull request by linh2931**
Remove chain_pacemaker and qc_chain related files and tests. Do the removal now so that we don't need to maintain the tests using qc_chain and chain_pacemaker while making changes to common files. 

Porting tests to work with the new implementation is tracked by https://github.com/AntelopeIO/leap/issues/2147.

Resolved https://github.com/AntelopeIO/leap/issues/2123.

### CODE REVIEW

File: libraries/chain/hotstuff/test/hotstuff_tools.cpp [hunk 1]
```diff
@@ -1,63 +1,14 @@
-#include <boost/test/unit_test.hpp>
+#define BOOST_TEST_MODULE hotstuff
+
```
Comment (heifner) [inline libraries/chain/hotstuff/test/hotstuff_tools.cpp:2]: `hotstuff_tools.cpp` seems like an odd name for this file. Maybe rename to `hotstuff_misc_tests.cpp`

**Review by heifner — CHANGES_REQUESTED** [review]

File: libraries/chain/hotstuff/test/hotstuff_tools.cpp (hunk 1 above)
Comment (linh2931) [reply libraries/chain/hotstuff/test/hotstuff_tools.cpp:2]: Changed to `finality_misc_tests.cpp` for eventual renaming `hotstuff` directory to `finality` directory.

**Review by heifner — APPROVED** [review]

**Comment by greg7mdp** [comment]
lgtm, but I'm wondering if before removing all the tests that existed for the qc_chain implementation, we should not at least review them, and convert those which make sense for our new implementation of IF?

**Comment by heifner** [comment]
> lgtm, but I'm wondering if before removing all the tests that existed for the qc_chain implementation, we should not at least review them, and convert those which make sense for our new implementation of IF?

That is what #2147 is for.

**Review by greg7mdp — APPROVED** [review]

**Comment by linh2931** [comment]
The adapting of the tests will be done by https://github.com/AntelopeIO/leap/issues/2147 (It will be a significant work).  The tests being removed do not cover any new implementation; removing them does not reduce our existing test coverage but saves us from updating them just for compile whenever we change some common data types.

**Comment by ericpassmore** [comment]
Note:start
group: IF
category: TEST
summary: Remove original pacemaker code and update tests to match current working code. 
Note: end

**Pull request merged**